### DISCUSSION TITLE
DISCUSSION: Change default rule comparison mode (contains --> equals)

### DISCUSSION THREAD

**Discussion by Dominic-Marcelino** [directus/directus#13560] [Feature Requests]
Hi everyone,

right now the default comparator for rules on string fields is `contains`.
For more strict permissions i'd suggest to use `equals` by default.

I think right now it's easy to oversee that for text fields a less strict comparator is used than for other types. Also I think that people will more likely notice when they used `equals` but need `contains` than opposite.

Thanks! 
Dominic

**Comment by rijkvanzanten** [comment]
Interesting! The reason it's _contains_ by default now is that it's the same component that's used in the search bar of the layouts, which should probably stay on "contains" for ease of use 🤔

**Reply by Dominic-Marcelino** [reply]
That's true! I think ideally it would be "contains" for search but "equals" for permissions.
Have no dev-setup of directus currently, but from what I saw in the codebase it should be more or less doable 🤔

**Reply by rijkvanzanten** [reply]
Should be doable! We'd have to add a new prop to that component to effectively say "Start with contains/equals" and set that prop based on the location it's used

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
